app/mailers/producer_mailer.rb

```diff
@@ -2,28 +2,31 @@ class ProducerMailer < Spree::BaseMailer
   include I18nHelper
 
   def order_cycle_report(producer, order_cycle)
-    @producer = producer
-    @coordinator = order_cycle.coordinator
-    @order_cycle = order_cycle
-    line_items = line_items_from(@order_cycle, @producer)
-    @grouped_line_items = line_items.group_by(&:product_and_full_name)
-    @receival_instructions = @order_cycle.receival_instructions_for @producer
-    @total = total_from_line_items(line_items)
-    @tax_total = tax_total_from_line_items(line_items)
-
-    I18n.with_locale valid_locale(@producer.owner) do
-      order_cycle_subject = I18n.t('producer_mailer.order_cycle.subject', producer: producer.name)
-      subject = "[#{Spree::Config.site_name}] #{order_cycle_subject}"
-
-      return unless has_orders?(order_cycle, producer)
-
-      mail(
-        to: @producer.contact.email,
-        from: from_address,
-        subject: subject,
-        reply_to: @coordinator.contact.email,
-        cc: @coordinator.contact.email
-      )
+    unscoping do
```

Comment:
The code inside this unscoping should be extracted to a method. this method is already too long.

Maybe a better name would be   with_unscoped_products_and_variants?

Reply:
Agree. This is totally awful.

Reply:
I've just run into a similar problem in another PR, and applying this `with_unscoped_products_and_variants` technique fixed the issue...